### PR TITLE
Added a Bun egg

### PR DESCRIPTION
**Pull request by mratox**
# Description
Added a Bun egg.

Resolves https://github.com/parkervcp/eggs/issues/2420

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [X] Does your submission pass tests (server is connectable)?
2. [X] Does your egg use a custom docker image?
    * [X] Have you tried to use a generic image?
    * [X] Did you PR the necessary changes to make it work?
3. [X] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [X] You verify that the start command applied does not use a shell script
    * [X] If some script is needed then it is part of a current yolk or a PR to add one
6. [X] The egg was exported from the panel

### CODE REVIEW

File: generic/bun/egg-bun.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2023-08-27T01:47:23+02:00",
+    "name": "Bun",
+    "author": "mratox@protonmail.com",
+    "description": "Bun is an incredibly fast JavaScript runtime, transpiler bundler, and npm package manager all-in-one.",
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/mratox\/images:bun": "ghcr.io\/mratox\/images:bun"
```
Comment (parkervcp) [inline generic/bun/egg-bun.json:13]: Please PR an image to my yolks repo. You will need to configure it in the build system as well

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: generic/bun/egg-bun.json (hunk 1 above)
Comment (mratox) [reply generic/bun/egg-bun.json]: Done

**Comment by QuintenQVD0** [comment]
I update the egg to use proper bash indentation, as is used in our example scripts. and also the user can not be allowed to set the startup cmd themselves as it is a security risk. And I added the image name as how they will be build

**Comment by mratox** [comment]
The required changes has been implemented and the docker image in the yolks has been approved.

**Comment by parkervcp** [comment]
Have you tried with the the images from the yolks?

**Comment by mratox** [comment]
Yes, i have tried the egg with yolks images, everything is working.

**Review by parkervcp — APPROVED** [review]

**Comment by parkervcp** [comment]
thanks for the work

**Pull request merged**